momoko/connection.py

```diff
@@ -30,6 +30,12 @@
 from tornado.ioloop import IOLoop
 from tornado.concurrent import chain_future, Future
 
+try:
+    from tornado.concurrent import future_set_exc_info
+except ImportError:
+    def future_set_exc_info(future, exc_info):
+        future.set_exc_info(exc_info)
```

Comment:
This line assumes the future is a `Tornado.concurrent.future`, but [this line](https://github.com/tornadoweb/tornado/blob/master/tornado/concurrent.py#L370) suggests that the Future can be an `asyncio.Future` (as is my case), so `.set_exc_info()` does not exist, you probably want to check what type of future this is an use [.set_exception()](https://docs.python.org/3/library/asyncio-task.html#asyncio.Future.set_exception) if it is an asyncio one.